kernel/device.c

```diff
@@ -112,7 +112,7 @@ const struct device *z_impl_device_get_binding(const char *name)
 	/* A null string identifies no device.  So does an empty
 	 * string.
 	 */
-	if ((name == NULL) || (*name == 0)) {
+	if ((name == NULL) || (*name == 0U)) {
 		return NULL;
```

Comment:
Wouldn’t '\0' be the most appropriate here since it’s a character and not an integer?

Reply:
It would also be more consistent with the place you’re changing in fatal.c

Reply:
Seems like you missed this comment? I think the signedness of `char` is also architecture dependent, which is why `\0` would also be better.